core/trino-main/src/main/java/io/trino/sql/planner/planprinter/JsonRenderer.java

```diff
@@ -60,7 +62,7 @@ private JsonRenderedNode renderJson(PlanRepresentation plan, NodeRepresentation
     {
         private final String id;
         private final String name;
-        private final String identifier;
+        private final Map<String, String> identifier;
         private final List<TypedSymbol> outputs;
         private final String details;
```

Comment:
Can we add a change in another subsequent commit to make this `List<String> details` and avoid new-line separated string in the json while keeping it the same in the text plan ?

Reply:
Not exactly sure if its going to help at all in terms of parsing the JSON plan.

Reply:
A newline separated string within a json structure looks like an anti-pattern to me, a list of things seems more appropriate. Ideally details could be even more structured, but that would increase scope of changes.